docs/notes.md

```diff
@@ -603,4 +603,254 @@ sends a pair \\((L\_j,R\_j)\\) of points at each step \\(j = k\dots1\\). An
 additional and final step involves sending a pair of scalars
 \\((a^{(0)}\_0,b^{(0)}\_0)\\) and checking the final relation directly.
 
-[bulletproofs_paper]: https://eprint.iacr.org/2017/1066.pdf
+Aggregated Range Proof
+======================
+
+We want to create an aggregated range proof for `m` values that is more efficient to create and verify than `m` individual range proofs.
+
+The aggregation protocol is a multi-party computation protocol, involving `m` parties (one party per value) and one dealer, where the parties don't reveal their secrets to each other. The parties share their commitments with the dealer, and the dealer generates and returns challenge variables. The parties then share their proof shares with the dealer, and the dealer combines their shares to create an aggregated proof. 
+
+The Bulletproofs paper outlines two versions of multi-party computation aggregation - one with a constant number of rounds but communication that is linear in both `m` and the binary encoding of the range, and one with a logarithmic number of rounds and communication that is only linear in `m`. We chose to implement the first version because the two versions don't differ significantly in proof size, and the first version is more straightforward.
```

Comment:
I think it's better not to say "don't differ significantly" because resulting proof sizes are exactly the same.

Rather than giving the asymptotics on the number of rounds and size of communication, we could say that the difference is whether the inner-product proof is performed by the dealer (which requires sending the vectors used for the inner-product), or whether the inner-product proof is performed using MPC (which sends less data, but requires one round for each iteration of the inner-product protocol).

From this point of view I think it's a lot more clear why the choice we made is the right one, because even for 64-bit ranges each party's vectors are only a few KB, which isn't worth doing extra round trips.

Reply:
👍